SimG4CMS/Calo/src/HGCalSD.cc

```diff
@@ -213,10 +216,19 @@ uint32_t HGCalSD::setDetUnitId(const G4Step* aStep) {
     if ((diff > tol) || (!valid1))
       pid = "HGCalError";
     auto partn = hgcons_->waferTypeRotation(hid1.layer(), hid1.waferU(), hid1.waferV(), false, false);
-    edm::LogVerbatim(pid) << "CheckID " << HGCSiliconDetId(id) << " input position: (" << hitPoint.x() / CLHEP::cm
-                          << ", " << hitPoint.y() / CLHEP::cm << "); position from ID (" << xx << ", " << xy.second
-                          << ") distance " << diff << " Valid " << valid1 << " Wafer type|rotation " << partn.first
-                          << ":" << partn.second << " CassetteShift " << cshift;
+    int indx = HGCalWaferIndex::waferIndex(layer, hid1.waferU(), hid1.waferV());
+    double phi = std::atan2(hitPoint.y(), hitPoint.x());
+    edm::LogVerbatim(pid) << "CheckID " << HGCSiliconDetId(id) << " Layer:Module:Cell:ModuleLev " << layer << ":"
+                          << module << ":" << cell << ":" << moduleLev << " SimWt:history " << useSimWt_ << ":"
+                          << touch->GetHistoryDepth() << ":" << levelT1_ << ":" << levelT2_ << " input position: ("
+                          << hitPoint.x() / CLHEP::cm << ", " << hitPoint.y() / CLHEP::cm << ":" << convertRadToDeg(phi)
```

Comment:
If you don't shield the log message and related computations into some `EDM_ML_DEBUG`, you could still save some computation in normal running by letting `phi` being calculated only inside the `LogVerbatim`:
```suggestion
    edm::LogVerbatim(pid) << "CheckID " << HGCSiliconDetId(id) << " Layer:Module:Cell:ModuleLev " << layer << ":"
                          << module << ":" << cell << ":" << moduleLev << " SimWt:history " << useSimWt_ << ":"
                          << touch->GetHistoryDepth() << ":" << levelT1_ << ":" << levelT2_ << " input position: ("
                          << hitPoint.x() / CLHEP::cm << ", " << hitPoint.y() / CLHEP::cm << ":"
                          << convertRadToDeg([phi](std::atan2(hitPoint.y(), hitPoint.x())))
```